crates/viewer/re_blueprint_tree/src/blueprint_tree.rs

```diff
@@ -576,11 +642,111 @@ impl BlueprintTree {
             ItemContext::BlueprintTree {
                 filter_session_id: self.filter_state.session_id(),
             },
-            &response,
+            response,
             SelectionUpdateBehavior::UseSelection,
         );
-        self.scroll_to_me_if_needed(ui, &item, &response);
-        ctx.handle_select_hover_drag_interactions(&response, item, true);
+        self.scroll_to_me_if_needed(ui, &item, response);
+        ctx.handle_select_hover_drag_interactions(response, item.clone(), true);
+
+        self.handle_range_selection(ctx, blueprint_tree_data, item, response);
+    }
+
+    /// Handle setting/extending the selection based on shift-clicking.
+    fn handle_range_selection(
+        &mut self,
+        ctx: &ViewerContext<'_>,
+        blueprint_tree_data: &BlueprintTreeData,
+        item: Item,
+        response: &Response,
+    ) {
+        // Early out if we're not being clicked.
+        if !response.clicked() {
+            return;
+        }
+
+        let modifiers = ctx.egui_ctx.input(|i| i.modifiers);
+
+        if modifiers.shift {
+            if self.last_clicked_item.is_some() {
+                let items_iterator = self
+                    .items_in_range(ctx, blueprint_tree_data, &item)
+                    .into_iter()
+                    .map(|item| {
+                        (
+                            item,
+                            Some(ItemContext::BlueprintTree {
+                                filter_session_id: self.filter_state.session_id(),
+                            }),
+                        )
+                    });
+
+                if items_iterator.len() > 0 {
+                    if modifiers.command {
```

Comment:
TIL I learned about this subtle behavior being standard (it truly is, tried it)

Reply:
since this is going to show up in more places, maybe that would be a good highlevel utility on the selection state? aka `selection_state.shift_select(modifiers.command, items_iterator)` (not a good name)

Reply:
I'm going to invoke the rule of 3 here :)